### ISSUE TITLE
Split Out GUI from RenameOp Implementations

### ISSUE THREAD

**Issue by edwardrowe** [redbluegames/unity-mulligan-renamer#64]
The GUI code is getting intermingled with the operations themselves, mainly with the ReplaceCharactersOperation.

To fix, we should pull out GUI into a separate class. BaseRenameOperation would implement it, using the DrawGUI method it already has.

It may be tricky to figure out how to bind these to the various RenameOp instances, but it can be done.

**Comment by edwardrowe** [comment]
When these are bound, they should be passed the colors for Add, Delete, Modify, and Replace instead of each operation storing them.

**Comment by edwardrowe** [comment]
Also as you do this, consider a framework for working on copies of the Model + View rather than working on "this" everywhere. So basically:
```
OnGui
{
    var preGUICopy = new Operation(this);
    var postGUICopy = new Operattion(this);
    postGUICopy.X = Draw(preGUICopy.X);
    ...
    this.CopyValuesFrom(postGUICopy);
}
```

**Comment by edwardrowe** [comment]
Fixed in develop-1.3.0
